pkgs/os-specific/linux/raspberrypi-eeprom/default.nix

```
@@ -24,18 +24,18 @@ stdenvNoCC.mkDerivation rec {
   '';
 
   installPhase = ''
-    mkdir -p $out/bin $out/share/rpi-eeprom
+    mkdir -p "$out/bin" "$out/lib/firmware/raspberrypi/bootloader-"{2711,2712}
 
-    cp rpi-eeprom-config rpi-eeprom-update rpi-eeprom-digest $out/bin
-    cp -r firmware/{beta,critical,old,stable} $out/share/rpi-eeprom
-    cp -P firmware/default firmware/latest $out/share/rpi-eeprom
+    cp rpi-eeprom-config rpi-eeprom-update rpi-eeprom-digest "$out/bin"
+    cp -rP firmware-2711/{beta,critical,default,latest,stable,old} "$out/lib/firmware/raspberrypi/bootloader-2711"
+    cp -rP firmware-2712/{beta,critical,default,latest,stable}     "$out/lib/firmware/raspberrypi/bootloader-2712"
   '';
 
   fixupPhase = ''
     patchShebangs $out/bin
     for i in rpi-eeprom-update rpi-eeprom-config; do
       wrapProgram $out/bin/$i \
-        --set FIRMWARE_ROOT $out/share/rpi-eeprom \
+        --set FIRMWARE_ROOT "$out/lib/firmware/raspberrypi/bootloader" \
```

```suggestion
        --set FIRMWARE_ROOT "$out/lib/firmware/raspberrypi" \
```

The target didn't seem to exist anyway. Unless this is intended to be a pattern/prefix, then i think referencing the parent dir is better?

---

Yes, this is indeed a prefix. Here is the code: https://github.com/raspberrypi/rpi-eeprom/blob/v2023.10.30-2712/rpi-eeprom-update#L407.